### DISCUSSION TITLE
Can nextJs export individual page among multiple dynamic pages?

### DISCUSSION THREAD

**Discussion by rockyip** [vercel/next.js#39011] [Help] [answered]
Hello, I am trying to create a landing CMS with strapi and nextJS, which can create multiple landing pages(one single page only)
It is good that nextJs can generate dynamic routes for each landing if I use **next start**

However, I want to export them individually so that I can copy those files to host on different servers
I have tried  **next build** and **next export** with **exportpathMap**, but it still build and export all pages at once.

Let's say, in my CMS there is 1000 landing pages using the same template pages/[slug].js with different images,
I want to click an export button and trigger nextJs to build and export one selected page only,
so that the out directory only contains the files of that selected page.
Is it possible?

**Comment by Babbili** [comment, marked as answer]
@rockyip you can assign a new attribute to each page's data in the CMS, lets call this attribute `isPublished` which is a `boolean` set it as `true` for the page you want to publish and as `false` for the page you don't want to publish.
 When fetching data from the CMS in `[slug].js` file, return only the pages that has `isPublished === true` then pass it to the Page Component as a prop.
example
```jsx
//    pages/[slug].js

function Page({ publishedData }) {
    console.log('publish pages', publishedData)

    return(
        <h1>{ publishedData.title }</h1>
    )
}
export default Page

export default async function getStaticProps( context ) {
    const slug = context.slug
    let res = await fetch(`https://examplecmsapi.com/data?${slug}`)
    let data = res.json()
    const publishedData = data.filter(data => data.isPublished === true )

    return{
        props: { publishedData }
    }
}
```

**Comment by krckyboy** [comment]
> Hello, I am trying to create a landing CMS with strapi and nextJS, which can create multiple landing pages(one single page only) It is good that nextJs can generate dynamic routes for each landing if I use **next start**
> 
> However, I want to export them individually so that I can copy those files to host on different servers I have tried **next build** and **next export** with **exportpathMap**, but it still build and export all pages at once.
> 
> Let's say, in my CMS there is 1000 landing pages using the same template pages/[slug].js with different images, I want to click an export button and trigger nextJs to build and export one selected page only, so that the out directory only contains the files of that selected page. Is it possible?

I'm sorry for this offtopic, but I'm wondering how did you handle assets for each landing page, considering that chunks and static files are in parent directory in `out` by default? I assume you have some kind of CI/CD for deploying each landing page separately, in a standalone fashion? Did you use custom webpack config so each page has all assets in the same directory level, per page folder?

Ideally what I want to achieve is that once I build landing pages, each landing page has its own copy of assets.
